app/org/sagebionetworks/bridge/services/ConsentPdf.java

```diff
@@ -106,11 +106,17 @@ private String createSignedDocument() {
         String username = Jsoup.clean(consentSignature.getName(), Whitelist.none());
         
         if (studyConsentAgreement.contains("<html")) {
-            // proceed as we used to
+            // legacy format, proceed as we used to. Note that once phone number was added to the system, 
+            // we didn't check for a null email, so this has been added after manual repair of consents 
+            // exposed this bug.
             String userEmail = signer.getEmail();
             String html = studyConsentAgreement.replace("@@name@@", username);
             html = html.replace("@@signing.date@@", signingDate);
-            html = html.replace("@@email@@", userEmail);
+            if (signer.getEmail() != null && Boolean.TRUE.equals(signer.getEmailVerified())) {
+                html = html.replace("@@email@@", userEmail);
+            } else {
+                html = html.replace("@@email@@", "");
+            }
```

Comment:
Was this intended to be part of this pull request? If so, should we add unit tests for this?

Reply:
Yes, integration tests run against my consents as fixed locally, ran a text against a legacy document when there was no email address, and this generates an NPE. (I don't think any of these legacy documents are in production.) Added a test for this.